Slim/Control/Commands.pm

```diff
@@ -3404,6 +3402,20 @@ sub _playlistXtracksCommand_parseSearchTerms {
 		}
 	}
 
+	# Check for 'sort' after iterating other keys, so that it takes precedence...
+	if (defined($terms->{'sort'})) {
+		my $value = $terms->{'sort'};
+
+		if ($value eq 'artflow' || $value eq 'yearartistalbum' || $value eq 'yearalbum') {
+			# If its an album sort where year takes precendence, then sort by year first
+			$sort = $albumYearSort;
+		} elsif ($value ne 'artistalbum' && $value ne 'new' && $value ne 'random') {
+			# Only use sort value if it is **not** an album sort.
+			# TODO: Better way to check this?
```

Comment:
What did you want to improve here?

Reply:
Well is there a way to detect that the sort is an SQL style sort. i.e. not explicty that its not an album-sort. 

The `else` part is setting the sort to whetever `sort` is in `$what`. For albums we only really care about the year version, as the others are handled elsewhere. It just seemed a little too album-specific to be checking if its not one of these. So, instead it would be nicer if it was something like:

```
if ($value eq 'artflow' || $value eq 'yearartistalbum' || $value eq 'yearalbum') {
} else if ($value is a valid sql(?) sort criteria) {
}
```

Reply:
There certainly could be something smarter. But I think it's good enough. Just remove the TODO comment :-).

I tried to understand whether any other caller would submit a `sort` value. But it might actually be more of a left over from when this code was used by other commands, too.